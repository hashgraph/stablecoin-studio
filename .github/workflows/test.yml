name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test