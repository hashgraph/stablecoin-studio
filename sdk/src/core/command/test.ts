/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
