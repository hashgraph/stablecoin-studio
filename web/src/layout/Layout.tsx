import { Box, Flex } from '@chakra-ui/react';
import type { ReactNode } from 'react';
import Sidebar from './sidebar/Sidebar';
import Topbar from './topbar/Topbar';

interface LayoutProps {
	children: ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
	return (
		<Box data-testid='layout' h='100vh' w='100%'>
			<Topbar />

			<Flex transition='transform 0.3s, width 0.3s' h='calc(100vh - 64px)'>
				<Sidebar />
				<Box w='100%' py={9} px={10} minH='100%' bgColor='brand.gray5'>
					{children}
				</Box>
			</Flex>
		</Box>
	);
};

export default Layout;
