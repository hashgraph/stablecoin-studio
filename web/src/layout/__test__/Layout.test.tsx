import Layout from '../Layout';
import { render } from '../../test/';

const layoutProps = {
	children: <span data-testid='layout-children'>TESTING</span>,
};

describe(`<${Layout.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<Layout {...layoutProps} />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has a topbar', () => {
		const component = render(<Layout {...layoutProps} />);

		const topbar = component.getByTestId('topbar');
		expect(topbar).toBeInTheDocument();
	});

	test('should has a sidebar', () => {
		const component = render(<Layout {...layoutProps} />);

		const topbar = component.getByTestId('sidebar');
		expect(topbar).toBeInTheDocument();
	});

	test('should render a children', () => {
		const component = render(<Layout {...layoutProps} />);

		const children = component.getByTestId('layout-children');
		expect(children).toBeInTheDocument();
	});
});
