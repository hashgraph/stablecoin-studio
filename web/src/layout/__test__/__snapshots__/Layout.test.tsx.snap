// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1jrkj43"
    data-testid="layout"
  >
    <div
      class="css-9go0lu"
      data-testid="topbar"
    >
      <div
        class="css-gut43g"
      >
        <img
          class="chakra-image css-z4bd0k"
          data-testid="topbar-logo"
          src="hedera-hbar-logo.svg"
        />
      </div>
      <div
        class="css-1grq7v1"
      />
      <div
        class="css-1gacwoi"
      >
        <div
          class="css-166re2f"
        >
          <div
            class="css-1hg55h3"
            data-testid="coin-dropdown"
          >
            <div
              class="chakra-stack css-1h0sumw"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                data-testid="form_control"
                role="group"
              >
                <div
                  class="css-frrdvf"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    class=" css-17n5zh"
                  >
                    <div
                      class="css-3y2vuh"
                    >
                      <span
                        class="chakra-text css-4pvr19"
                        data-testid="select-placeholder"
                      >
                        Choose Stable coin
                      </span>
                      <div
                        class=" css-svnjlw"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-describedby="react-select-2-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=" css-146k1ar-input"
                          id="field-:r0:"
                          role="combobox"
                          spellcheck="false"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-hfbj6y"
                    >
                      <div
                        class="css-1aabyhj"
                      >
                        <svg
                          class="chakra-icon css-zsadqz"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="0 0 256 256"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="256"
                            width="256"
                          />
                          <polyline
                            fill="none"
                            points="208 96 128 176 48 96"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="16"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="coin-dropdown"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-17tfhss"
            type="button"
          >
            <svg
              class="chakra-icon css-ky7im0"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="40"
                x2="216"
                y1="128"
                y2="128"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="128"
                x2="128"
                y1="40"
                y2="216"
              />
            </svg>
          </button>
        </div>
        <div
          class="css-166re2f"
          data-testid="topbar-right"
        >
          <div
            class="chakra-stack css-d17m5h"
          >
            <div
              class="chakra-stack css-jp20jv"
            >
              <div
                class="css-1m8iww1"
              >
                <p
                  class="chakra-text css-o1au3i"
                  data-testid="topbar-right-account"
                >
                  0.0.48450590
                </p>
              </div>
              <p
                class="chakra-text css-gegx7f"
                data-testid="topbar-right-network"
              >
                testnet
              </p>
            </div>
            <img
              alt="HashPack"
              class="chakra-image css-1lzenv"
              src="hashpackLogo.png"
            />
          </div>
          <div
            class="css-dv2lk7"
          />
          <div
            class="css-1a9f5vk"
          >
            <svg
              class="chakra-icon css-dee54c"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="128"
                x2="128"
                y1="48"
                y2="124"
              />
              <path
                d="M176,54.2a88,88,0,1,1-96,0"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-1jaqult"
    >
      <div
        class="css-6i4ah3"
        data-testid="sidebar"
      >
        <div
          class="css-17lnxuq"
        >
          <div
            class="css-18m8sk"
            data-testid="sidebar-option-Coin"
          >
            <svg
              class="chakra-icon css-wrbp2a"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <ellipse
                cx="128"
                cy="104"
                fill="none"
                rx="104"
                ry="48"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="128"
                x2="128"
                y1="152"
                y2="200"
              />
              <path
                d="M24,104v48c0,24,40,48,104,48s104-24,104-48V104"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="192"
                x2="192"
                y1="142.1"
                y2="190.1"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="64"
                x2="64"
                y1="142.1"
                y2="190.1"
              />
            </svg>
            <p
              class="chakra-text css-h0pfk0"
              data-testid="sidebar-option-title"
            >
              Operations
            </p>
          </div>
          <div
            class="css-18m8sk"
            data-testid="sidebar-option-Users"
          >
            <svg
              class="chakra-icon css-wrbp2a"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="88"
                cy="108"
                fill="none"
                r="52"
                stroke="currentColor"
                stroke-miterlimit="10"
                stroke-width="16"
              />
              <path
                d="M155.4,57.9A54.5,54.5,0,0,1,169.5,56a52,52,0,0,1,0,104"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M16,197.4a88,88,0,0,1,144,0"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M169.5,160a87.9,87.9,0,0,1,72,37.4"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <p
              class="chakra-text css-h0pfk0"
              data-testid="sidebar-option-title"
            >
              Role management
            </p>
          </div>
          <div
            class="css-18m8sk"
            data-testid="sidebar-option-Gear"
          >
            <svg
              class="chakra-icon css-wrbp2a"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="48"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M183.7,65.1q3.8,3.5,7.2,7.2l27.3,3.9a103.2,103.2,0,0,1,10.2,24.6l-16.6,22.1s.3,6.8,0,10.2l16.6,22.1a102.2,102.2,0,0,1-10.2,24.6l-27.3,3.9s-4.7,4.9-7.2,7.2l-3.9,27.3a103.2,103.2,0,0,1-24.6,10.2l-22.1-16.6a57.9,57.9,0,0,1-10.2,0l-22.1,16.6a102.2,102.2,0,0,1-24.6-10.2l-3.9-27.3q-3.7-3.5-7.2-7.2l-27.3-3.9a103.2,103.2,0,0,1-10.2-24.6l16.6-22.1s-.3-6.8,0-10.2L27.6,100.8A102.2,102.2,0,0,1,37.8,76.2l27.3-3.9q3.5-3.7,7.2-7.2l3.9-27.3a103.2,103.2,0,0,1,24.6-10.2l22.1,16.6a57.9,57.9,0,0,1,10.2,0l22.1-16.6a102.2,102.2,0,0,1,24.6,10.2Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <p
              class="chakra-text css-h0pfk0"
              data-testid="sidebar-option-title"
            >
              Stable coin details
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-l3typs"
      >
        <span
          data-testid="layout-children"
        >
          TESTING
        </span>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
